### PR TITLE
Appview blocks

### PR DESCRIPTION
**Pull request by dholms**
Implements blocks on the appview

### CODE REVIEW

File: packages/dev-env/src/network.ts [hunk 1]
```diff
@@ -75,6 +75,13 @@ export class TestNetwork extends TestNetworkNoAppView {
     throw new Error(`Sequence was not processed within ${timeout}ms`)
   }
 
+  async processAll(timeout?: number) {
+    await this.pds.ctx.backgroundQueue.processAll()
+    if (!this.bsky) return
+    await this.processFullSubscription(timeout)
+    await this.bsky.ctx.backgroundQueue.processAll()
+  }
```
Comment (devinivy) [inline packages/dev-env/src/network.ts:83]: Yupyup, been wanting to do exactly this 💪

File: packages/bsky/src/services/feed/index.ts [hunk 2]
```diff
@@ -454,3 +462,44 @@ export class FeedService {
...
+  if (!str) return str
+  const encoder = new TextEncoder()
+  const utf8 = encoder.encode(str)
+  if (utf8.length > length) {
+    const decoder = new TextDecoder('utf-8', { fatal: false })
+    const truncated = utf8.slice(0, length)
+    return decoder.decode(truncated).replace(/\uFFFD$/, '')
+  }
+  return str
+}
```
Comment (devinivy) [inline packages/bsky/src/services/feed/index.ts:476]: I _think_ we can get rid of this now that all clients are compatible with the lex refactor.

File: packages/bsky/src/services/indexing/index.ts [hunk 3]
```diff
@@ -54,6 +56,7 @@ export class IndexingService {
       profile: Profile.makePlugin(this.db, backgroundQueue),
       list: List.makePlugin(this.db, backgroundQueue),
       listItem: ListItem.makePlugin(this.db, backgroundQueue),
+      block: Block.makePlugin(this.db, backgroundQueue),
```
Comment (devinivy) [inline packages/bsky/src/services/indexing/index.ts:59]: I think we should remove records from the `actor_block` table during `unindexActor()` too.

File: packages/bsky/tests/views/blocks.test.ts [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+    const resDan = await agent.api.app.bsky.feed.getTimeline(
+      { limit: 100 },
+      { headers: await network.serviceHeaders(dan) },
+    )
+    expect(
+      resDan.data.feed.some((post) => post.post.author.did === carol),
+    ).toBeFalsy()
+  })
+
+  // it('strips blocked users out of getPopular', async () => {
```
Comment (devinivy) [inline packages/bsky/tests/views/blocks.test.ts:138]: Does this remain commented out intentionally?

**Review by devinivy — APPROVED** [review]
Had a few minor thoughts, but this is looking great 🙏

File: packages/bsky/src/services/feed/index.ts (hunk 2 above)
Comment (dholms) [reply packages/bsky/src/services/feed/index.ts:476]: ah cool cool checks out 👍 

i couldn't quite remember when we'd added this in & erred on the side of caution

File: packages/bsky/src/services/indexing/index.ts (hunk 3 above)
Comment (dholms) [reply packages/bsky/src/services/indexing/index.ts:59]: oh yup totally. good call 👍

File: packages/bsky/tests/views/blocks.test.ts (hunk 4 above)
Comment (dholms) [reply packages/bsky/tests/views/blocks.test.ts:138]: yea removed that method from the appview - i'll just remove this test

**Pull request merged**